File: control-assurance-sp/src/types/ClWorker.ts
import { IEntity } from "./Entity";


export interface ICLWorker  extends IEntity {

    CLCaseId?: number;
    Stage?: string;
    WorkerChangeLog?: string;
    WorkerNumber?: number;
    OnbContractorFirstname?: string;
    OnbContractorSurname?: string;
    OnbStartDate?: Date;
    OnbEndDate?: Date;
    OnbDayRate?: number;
    OnbContractorHomeAddress?: string;
    OnbContractorEmail?: string;
    OnbContractorPhone?: string;
    PurchaseOrderNum?: string;
    OnbSecurityClearanceId?: number;

    OnbContractorTitleId?: number;
    OnbContractorGender?: string;
    OnbContractorDob?: Date;
    OnbContractorNINum?: string;
    OnbContractorPostCode?: string;
    OnbWorkingDayMon?: boolean;
    OnbWorkingDayTue?: boolean;
    OnbWorkingDayWed?: boolean;
    OnbWorkingDayThu?: boolean;
    OnbWorkingDayFri?: boolean;
    OnbWorkingDaySat?: boolean;
    OnbWorkingDaySun?: boolean;
    OnbDecConflictId?: number;
    OnbLineManagerUserId?: number;
    OnbLineManagerGradeId?: number;
    OnbLineManagerEmployeeNum?: string;
    OnbLineManagerPhone?: string;
    OnbWorkOrderNumber?: string;
    OnbRecruitersEmail?: string;


    BPSSCheckedById?: number;
    BPSSCheckedOn?: Date;
    POCheckedById?: number;
    POCheckedOn?: Date;
    ITCheckedById?: number;
    ITCheckedOn?: Date;
    UKSBSCheckedById?: number;
    UKSBSCheckedOn?: Date;
    PassCheckedById?: number;
    PassCheckedOn?: Date;
    ContractCheckedById?: number;
    ContractCheckedOn?: Date;
    EngagedChecksDone: boolean;

    ITSystemRef?:string;
    ITSystemNotes?:string;
    UKSBSRef?:string;
    UKSBSNotes?:string;

    EngPONumber?:string;
    EngPONote?:string;

    LeEndDate?: Date;
    LeContractorPhone?: string;
    LeContractorEmail?: string;
    LeContractorHomeAddress?: string;
    LeContractorPostCode?: string;
    LeContractorDetailsCheckedById?: number;
    LeContractorDetailsCheckedOn?: Date;
    LeITCheckedById?: number;
    LeITCheckedOn?: Date;
    LeUKSBSCheckedById?: number;
    LeUKSBSCheckedOn?: Date;
    LePassCheckedById?: number;
    LePassCheckedOn?: Date;

    SDSPdfStatus?: string;
    SDSPdfDate?: Date;
    SDSPdfName?: string;
    SDSPdfLastActionUser?: string;


}

export class CLWorker implements ICLWorker{ 
    public ID: number = 0;
    public Title: string = null;
    public CLCaseId?: number = null;
    public Stage?: string = null;
    public WorkerChangeLog?: string = null;
    public WorkerNumber?: number = null;
    public OnbContractorFirstname?: string = null;
    public OnbContractorSurname?: string = null;
    public OnbStartDate?: Date = null;
    public OnbEndDate?: Date = null;
    public OnbDayRate?: number = null;
    public OnbContractorHomeAddress?: string = null;
    public OnbContractorEmail?: string = null;
    public OnbContractorPhone?: string = null;
    public PurchaseOrderNum?: string = null;
    public OnbSecurityClearanceId?: number = null;

    public OnbContractorTitleId?: number = null;
    public OnbContractorGender?: string = null;
    public OnbContractorDob?: Date = null;
    public OnbContractorNINum?: string = null;
    public OnbContractorPostCode?: string = null;
    public OnbWorkingDayMon?: boolean = null;
    public OnbWorkingDayTue?: boolean = null;
    public OnbWorkingDayWed?: boolean = null;
    public OnbWorkingDayThu?: boolean = null;
    public OnbWorkingDayFri?: boolean = null;
    public OnbWorkingDaySat?: boolean = null;
    public OnbWorkingDaySun?: boolean = null;
    public OnbDecConflictId?: number = null;
    public OnbLineManagerUserId?: number = null;
    public OnbLineManagerGradeId?: number = null;
    public OnbLineManagerEmployeeNum?: string = null;
    public OnbLineManagerPhone?: string = null;

    public BPSSCheckedById?: number = null;
    public BPSSCheckedOn?: Date = null;
    public POCheckedById?: number = null;
    public POCheckedOn?: Date = null;
    public ITCheckedById?: number = null;
    public ITCheckedOn?: Date = null;
    public UKSBSCheckedById?: number = null;
    public UKSBSCheckedOn?: Date = null;
    public PassCheckedById?: number = null;
    public PassCheckedOn?: Date = null;
    public ContractCheckedById?: number = null;
    public ContractCheckedOn?: Date = null;
    public EngagedChecksDone: boolean = null;

    public ITSystemRef?:string = "";
    public ITSystemNotes?:string = "";
    public UKSBSRef?:string = "";
    public UKSBSNotes?:string = "";

    public LeEndDate?: Date = null;
    public LeContractorPhone?: string = null;
    public LeContractorEmail?: string = null;
    public LeContractorHomeAddress?: string = null;
    public LeContractorPostCode?: string = null;
    public LeContractorDetailsCheckedById?: number = null;
    public LeContractorDetailsCheckedOn = null;
    public LeITCheckedById?: number = null;
    public LeITCheckedOn?: Date = null;
    public LeUKSBSCheckedById?: number = null;
    public LeUKSBSCheckedOn?: Date = null;
    public LePassCheckedById?: number = null;
    public LePassCheckedOn?: Date = null;

    public SDSPdfStatus?: string = null;
    public SDSPdfDate?: Date = null;
    public SDSPdfName?: string = null;
    public SDSPdfLastActionUser?: string = null;


    constructor() {
        
    }


}

File: ControlAssuranceAPI/Models/CLWorker.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class CLWorker
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public int? CLCaseId { get; set; }

    public string? Stage { get; set; }

    public string? WorkerChangeLog { get; set; }

    public int? WorkerNumber { get; set; }

    public string? OnbContractorFirstname { get; set; }

    public string? OnbContractorSurname { get; set; }

    public DateTime? OnbStartDate { get; set; }

    public DateTime? OnbEndDate { get; set; }

    public decimal? OnbDayRate { get; set; }

    public string? OnbContractorHomeAddress { get; set; }

    public string? OnbContractorEmail { get; set; }

    public string? OnbContractorPhone { get; set; }

    public string? PurchaseOrderNum { get; set; }

    public int? OnbSecurityClearanceId { get; set; }

    public int? OnbContractorTitleId { get; set; }

    public DateTime? OnbContractorDob { get; set; }

    public string? OnbContractorNINum { get; set; }

    public string? OnbContractorPostCode { get; set; }

    public int? OnbDecConflictId { get; set; }

    public int? OnbLineManagerUserId { get; set; }

    public int? OnbLineManagerGradeId { get; set; }

    public string? OnbLineManagerEmployeeNum { get; set; }

    public string? OnbLineManagerPhone { get; set; }

    public bool? OnbWorkingDayMon { get; set; }

    public bool? OnbWorkingDayTue { get; set; }

    public bool? OnbWorkingDayWed { get; set; }

    public bool? OnbWorkingDayThu { get; set; }

    public bool? OnbWorkingDayFri { get; set; }

    public bool? OnbWorkingDaySat { get; set; }

    public bool? OnbWorkingDaySun { get; set; }

    public int? BPSSCheckedById { get; set; }

    public DateTime? BPSSCheckedOn { get; set; }

    public int? POCheckedById { get; set; }

    public DateTime? POCheckedOn { get; set; }

    public int? ITCheckedById { get; set; }

    public DateTime? ITCheckedOn { get; set; }

    public int? UKSBSCheckedById { get; set; }

    public DateTime? UKSBSCheckedOn { get; set; }

    public int? PassCheckedById { get; set; }

    public DateTime? PassCheckedOn { get; set; }

    public int? ContractCheckedById { get; set; }

    public DateTime? ContractCheckedOn { get; set; }

    public bool? EngagedChecksDone { get; set; }

    public DateTime? LeEndDate { get; set; }

    public string? LeContractorPhone { get; set; }

    public string? LeContractorEmail { get; set; }

    public string? LeContractorHomeAddress { get; set; }

    public string? LeContractorPostCode { get; set; }

    public int? LeContractorDetailsCheckedById { get; set; }

    public DateTime? LeContractorDetailsCheckedOn { get; set; }

    public int? LeITCheckedById { get; set; }

    public DateTime? LeITCheckedOn { get; set; }

    public int? LeUKSBSCheckedById { get; set; }

    public DateTime? LeUKSBSCheckedOn { get; set; }

    public int? LePassCheckedById { get; set; }

    public DateTime? LePassCheckedOn { get; set; }

    public int? ExtendedFromWorkerId { get; set; }

    public string? ITSystemRef { get; set; }

    public string? ITSystemNotes { get; set; }

    public string? UKSBSRef { get; set; }

    public string? UKSBSNotes { get; set; }

    public string? OnbWorkOrderNumber { get; set; }

    public string? OnbRecruitersEmail { get; set; }

    public string? EngPONumber { get; set; }

    public string? EngPONote { get; set; }

    public string? SDSPdfStatus { get; set; }

    public DateTime? SDSPdfDate { get; set; }

    public string? SDSPdfName { get; set; }

    public string? SDSPdfLastActionUser { get; set; }

    public int? OnbContractorGenderId { get; set; }

    public int? SDSCheckedById { get; set; }

    public DateTime? SDSCheckedOn { get; set; }

    public string? SDSNotes { get; set; }

    public string? CasePdfStatus { get; set; }

    public DateTime? CasePdfDate { get; set; }

    public string? CasePdfName { get; set; }

    public string? CasePdfLastActionUser { get; set; }

    public bool? Archived { get; set; }

    public DateTime? OnbSubmitDate { get; set; }

    public DateTime? LeMoveToArchiveDate { get; set; }

    public virtual CLCase? CLCase { get; set; }

    public virtual CLGender? CLGender { get; set; }

    public virtual PersonTitle? PersonTitle { get; set; }

    public virtual CLDeclarationConflict? CLDeclarationConflict { get; set; }

    public virtual CLStaffGrade? CLStaffGrade { get; set; }

    public virtual CLSecurityClearance? CLSecurityClearance { get; set; }
}
public enum PdfType { SDSPdf, CasePdf }
